common/src/main/java/bisq/common/util/ProtobufUtils.java

```diff
@@ -41,6 +41,25 @@ public static <E extends Enum<E>> E enumFromProto(Class<E> enumType, String name
         return result;
     }
 
+    public static <E extends Enum<E>> E enumFromProto(Class<E> enumType, String name, E fallBack) {
+        String info = "Enum type= " + enumType.getSimpleName() + "; name=" + name + "; fallBack=" + fallBack;
```

Comment:
The `name == null` check must be before the `if (name.endsWith("_UNSPECIFIED"))`.

Convert the enum class name to the protobuf prefix (instead at each caller):
```
String protobufEnumPrefix = StringUtils.capitalizeAll(enumType.getSimpleName()) + "_";
        String enumName = name.replace(protobufEnumPrefix,"");
        E result = Enums.getIfPresent(enumType, enumName).orNull();
```

Add that method to StringUtils: 
```
 public static String capitalizeAll(String value) {
        if (value == null || value.isEmpty()) {
            return value;
        } else {
            return value.toUpperCase();
        }
    }
```

The method with no fallback should throw an exception if no valid enum can be resolved.

Methods should be:
```
public static <E extends Enum<E>> E enumFromProto(Class<E> enumType, String name, E fallBack) {
        String info = "Enum type= " + enumType.getSimpleName() + "; name=" + name + "; fallBack=" + fallBack;
        if (name == null) {
            log.error("Enum name is null. We use the fallback value instead. {}", info);
            return fallBack;
        }
        
        if (name.endsWith("_UNSPECIFIED")) {
            log.warn("Unspecified enum. We use the fallback value instead. {}", info);
            return fallBack;
        }

        String protobufEnumPrefix = StringUtils.capitalizeAll(enumType.getSimpleName()) + "_";
        String enumName = name.replace(protobufEnumPrefix,"");
        E result = Enums.getIfPresent(enumType, enumName).orNull();
        if (result == null) {
            log.error("Enum could not be resolved. We use the fallback value instead. {}", info);
            return fallBack;
        }
        return result;
    }

// If we do not provide a fallback value we throw an exception
public static <E extends Enum<E>> E enumFromProto(Class<E> enumType, String name) {
        String info = "Enum type= " + enumType.getSimpleName() + "; name=" + name;
        checkNotNull(name, "Enum name must not be null. "+info);
        checkArgument(!name.endsWith("_UNSPECIFIED"), "Unspecified enum. "+info);
       
        String protobufEnumPrefix = StringUtils.capitalizeAll(enumType.getSimpleName()) + "_";
        String enumName = name.replace(protobufEnumPrefix,"");
        E result = Enums.getIfPresent(enumType, enumName).orNull();
        checkNotNull(result, "Enum could not be resolved. "+info);
        return result;
    }
```